service/static/js/rest_api.js

```diff
@@ -47,26 +66,59 @@ $(function () {
     function render_promotion_data(promotions, table) {
         let tableBody = $(`#${table} tbody`);
         tableBody.empty();  // Clear existing data
-
         promotions.forEach(promotion => {
             // Populate the table with the retrieved promotion data
+            let activeText = promotion.active_status ? "Deactivate" : "Activate";
             const row = `
-                <tr>
+                <tr id="row-${promotion.id}">
                     <td>${promotion.id}</td>
                     <td>${promotion.name}</td>
                     <td>${promotion.description}</td>
                     <td>${promotion.product_ids}</td>
                     <td>${promotion.start_date}</td>
                     <td>${promotion.end_date}</td>
-                    <td>${promotion.active_status ? 'Active' : 'Inactive'}</td>
+                    <td class="status-cell">${promotion.active_status ? 'Active' : 'Inactive'} 
+                        <button 
+                            class="btn btn-sm btn-toggle" 
+                            id="toggle-btn"
+                            data-id="${promotion.id}" 
+                            data-status="${promotion.active_status}">
+                            ${activeText}
+                        </button>
+                    </td>
                     <td>${promotion.created_by}</td>
                     <td>${promotion.updated_by}</td>
                     <td>${promotion.created_at}</td>
                     <td>${promotion.updated_at}</td>
                     <td>${JSON.stringify(promotion.extra)}</td>
-                </tr>
+                        
+            </tr>
             `;
             tableBody.append(row);
+
+        $("#promotion-data").on("click", ".btn-toggle", function () {
```

Comment:
This line will cause the problem it should be `#${table}` instead of "#promotion-data"

here is my idea comes from: 
I've modified the function to support separate tables for the search and retrieve tabs. This change introduces distinct tables for each tab, allowing us to maintain search results and retrieved data separately. I believe this approach is more logical, as it prevents data loss by avoiding the need to clear the table when switching between tabs.

Reply:
Sure, I have modified it.